app/src/main/java/at/bitfire/icsdroid/model/ViewModelUtils.kt

```diff
@@ -0,0 +1,22 @@
+package at.bitfire.icsdroid.model
+
+import android.content.Context
+import android.widget.Toast
+import androidx.annotation.IntDef
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+@Retention(AnnotationRetention.SOURCE)
+@IntDef(Toast.LENGTH_SHORT, Toast.LENGTH_LONG)
+annotation class ToastDuration
+
+suspend fun toastAsync(
```

Comment:
This seems unnecessarily confusing to me. I am sorry, that I said we could extract it to utils. It does not make sense. I think it's much better to simply trigger and handle cancelation in place when needed. Either in the screens  (preferably) or the view model if you think that's much cleaner.